src/test/kotlin/com/zepben/auth/client/ZepbenTokenFetcherTest.kt

```diff
@@ -0,0 +1,252 @@
+// Copyright 2022 Zeppelin Bend Pty Ltd
+// This file is part of zepben-auth.
+//
+// zepben-auth is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// zepben-auth is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with zepben-auth.  If not, see <https://www.gnu.org/licenses/>.
+
+package com.zepben.auth.client
+
+import com.zepben.auth.common.AuthException
+import com.zepben.auth.common.AuthMethod
+import com.zepben.auth.common.StatusCode
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.AfterEach
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import com.zepben.testutils.vertx.TestHttpServer
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.core.IsEqual.equalTo
+import org.mockito.Mockito.*
+import com.zepben.testutils.auth.TOKEN
+
+import java.net.http.HttpClient
+import java.net.http.HttpResponse
+
+
+internal class ZepbenTokenFetcherTest {
+    private var server: TestHttpServer? = null
+    private var port = 8080
+
+    private val client: HttpClient = mock(HttpClient::class.java)
+    private val response = mock(HttpResponse::class.java)
+
+    @BeforeEach
+    fun beforeEach() {
+        server = TestHttpServer()
+        port = server!!.listen()
+        doReturn(response).`when`(client).send(any(), any(HttpResponse.BodyHandler::class.java))
+    }
+
+    @AfterEach
+    fun afterEach() {
+        server?.close()
+    }
+
+    @Test
+    fun testCreateTokenFetcherSuccess() {
+        doReturn(StatusCode.OK.code).`when`(response).statusCode()
+        doReturn(
+            "{\"authType\": \"AUTH0\", \"audience\": \"test_audience\", \"issuer\": \"test_issuer\"}"
+        ).`when`(response).body()
+
+        val tokenFetcher = createTokenFetcher("https://testaddress", confClient = client, authClient = client)
+        verify(client, times(1)).send(any(), any(HttpResponse.BodyHandler::class.java))
```

Comment:
prefer blank line between code execution and test validation.

no need for `times(1)` as this is the default for `verify`

prefer mokito.kotlin `any<HttpResponse.BodyHandler>()`